aida/tutorial08/tutorial08.py

```diff
@@ -0,0 +1,213 @@
+            for word_label in word_labels:
+                word, label = word_label.split('_')
+                words.append(word_to_id[word])
+                labels_each.append(label_to_id[label])
+            sentences_train.append(words)
+            labels_train.append(labels_each)
+    file_test = file_names[1]
+    with open(file_test) as fp:
+        sentences_test = []
+        for line in fp:
+            sentence = []
+            words = line.strip().split()
+            for word in words:
+                if word in word_to_id.keys():
+                    sentence.append(word_to_id[word])
+                else:
+                    sentence.append(word_to_id['<UNK>'])
+            sentences_test.append(sentence)
+    return sentences_train, labels_train, sentences_test, word_to_id, label_to_id
+
+def create_id2label(label_to_id):
+    id_to_label = {}
+    for label, id in label_to_id.items():
+        id_to_label[id] = label
+    return id_to_label
+
+def obtain_onehot(sentence, token_to_id):
+    """ create onehot vector from one sentence
+
+    :return: onehot_vecs 
+    """
+    onehot_vecs = np.zeros([len(sentence), len(token_to_id)])
+    for i, word_id in enumerate(sentence):
+        onehot_vecs[i][word_id] += 1
+    return onehot_vecs
+
+def criterion(y_hat, y_ans):
+    """ compute loss
+    
+    :param y_hat: numpy.array, predict label
+    :param y_ans: numpy.array, answer label
+    :return: loss
+    """
+    loss = 0
+    for hat, ans in zip(y_hat, y_ans):
+        loss += 1/2 * np.linalg.norm(hat - ans)**2
+    return loss
+
+def plot_loss(losses):
+    """ plot train losses """
+    plt.xlabel('epoch')
+    plt.ylabel('loss')
+    plt.plot(range(len(losses)), losses)
+    plt.savefig('./tutorial08/loss.png')
+
+def decode(pred_sent, id_to_label):
+    """ convert id to label in prediction
+
+    :param pred_sent: predictions in one sentnece
+    :param id_to_label: dict
+    :return: labels
+    """
+    labels = [id_to_label[np.argmax(pred)] for pred in pred_sent]
+    return labels
+
+if __name__ == '__main__':
+    #file_train = './test/05-train-input.txt'
+    #file_test = './test/05-test-input.txt'
+    file_train = './data/wiki-en-train.norm_pos'
+    file_test = './data/wiki-en-test.norm'
+    sentences_train, labels_train, sentences_test, word_to_id, label_to_id = preprocess([file_train, file_test])
+    id_to_label = create_id2label(label_to_id)
+
+    # define params
+    learning_rate = 0.001
+    embed_size = len(word_to_id)
+    hidden_size = 50
+    output_size = len(label_to_id)
+    max_epoch = 20
+
+    rnn = RNN(learning_rate, word_to_id, label_to_id, embed_size, hidden_size, output_size)
+    rnn.fit(sentences_train, labels_train, max_epoch)
+    predictions = rnn.predict(sentences_test)
+
+    with open('./tutorial08/my-answer.pos', 'w') as fp:
+        for pred_sent in predictions:
+            labels = decode(pred_sent, id_to_label)
+            fp.write(f'{" ".join(labels)}\n')
```

Comment:
```suggestion
            fp.write(" ".join(labels) + '\n')
```

の方が見やすい気がします
`f'UNI:{word}'` の場面では f-strings の方が見やすいかもですが，好みだと思いますし（あとコーディング規約）

ちなみに `print(" ".join(labels), file=fp)` って書くこともできる気がします（遅いかもです）